src/server/raftkv.rs

```diff
@@ -201,7 +200,7 @@ where
         cb: Callback<CmdRes<E::Snapshot>>,
     ) -> Result<()> {
         let mut header = self.new_request_header(ctx.pb_ctx);
-        if ctx.pb_ctx.get_stale_read() && !ctx.start_ts.is_zero() {
+        if ctx.pb_ctx.get_stale_read() && (!ctx.start_ts.is_zero() || ctx.is_snap_for_gc) {
```

Comment:
Do you know originally why ctx.start_ts.is_zero() is not allowed?  I'm not sure if there's other changes needed to accommodate with this change.

Reply:
The original checks for ctx.start_ts.is_zero() is to avoid the casee: when TiKV start/restart, the safe_ts will be 0, and TiDB sends stale read with 0 start_ts.